Drivers/rc_receiver/Inc/sbus_receiver.hpp

```diff
@@ -1,10 +1,22 @@
 #ifndef SBUS_RECEIVER_HPP
 #define SBUS_RECEIVER_HPP
 
-#include "usart.h"
 #include "rcreceiver_datatypes.h"
 #include "sbus_defines.h"
 
+#ifdef UNIT_TEST
+    typedef struct __UART_HandleTypeDef{
+    }UART_HandleTypeDef;
+
+    typedef enum
+    {
+    } HAL_StatusTypeDef;
+    
+    HAL_StatusTypeDef HAL_UART_Receive_DMA(UART_HandleTypeDef *huart, uint8_t *pData, uint16_t Size);
+#else
+    #include "usart.h"
+#endif
```

Comment:
interesting, so the plan is to stub out all the hal functions here? It looks slightly messy especially since its in the actual file... would it be at all possible to have this in a different file? Like a UT_sbus_receiver.hpp file or something? Is that even possible? I'd like to know everyones thoughts...

Reply:
After some discussion: having a directory like Testing/Inc/ and Testing/Src/ to put the files related for each stub... for example since you are using a uart function, you would create the uart.h/c file (or whatever the actual HAL file is named) and add your stub function in here. Now people that want to use the same function can reference it from that file, if its not in there, they can append it to the bottom, and if the file doesn't exist they can make it.

Reply:
It looks like the fake function framework does something similar to what we discussed. It looks like fff is the real solution here (and it is more robust than just stubbing out a bunch of functions). I think the same concept still holds with fff though (putting the faked functions definitions in their own files, and compiling them in only if we are using a testing build)